openff/toolkit/topology/molecule.py

```diff
@@ -4634,121 +4661,102 @@ def from_qcschema(
-        >>> client = FractalClient()
+        >>> from qcportal import PortalClient
+        >>> client = PortalClient("https://api.qcarchive.molssi.org:443/")
         >>> offmol = Molecule.from_qcschema(
-        ...     client.query_molecules(molecular_formula="C7H12N2O4")[0]
+        ...     [*client.query_molecules(molecular_formula="C16H20N3O5")][-1]
         ... )
+        >>> offmol.to_hill_formula()
+        'C16H20N3O5'
 
         Get Molecule from a QCArchive optimization entry:
 
-        >>> from qcportal import FractalClient
-        >>> client = FractalClient()
-        >>> optds = client.get_collection(
-        ...     "OptimizationDataset",
-        ...     "SMIRNOFF Coverage Set 1"
-        ... )
-        >>> offmol = Molecule.from_qcschema(optds.get_entry('coc(o)oc-0'))
-
-        Same as above, but with conformer(s) from initial molecule(s) by
-        providing client to database:
-
-        >>> offmol = Molecule.from_qcschema(
-        ...     optds.get_entry('coc(o)oc-0'),
-        ...     client=client
+        >>> from qcportal import PortalClient
+        >>> client = PortalClient("https://api.qcarchive.molssi.org:443/")
+        >>> optimizations = client.get_dataset(
+        ...     dataset_type="optimization",
+        ...     dataset_name="SMIRNOFF Coverage Set 1",
         ... )
+        >>> offmol = Molecule.from_qcschema(optimizations.get_entry('coc(o)oc-0'))
+        >>> offmol.to_hill_formula()
+        'C3H8O3'
 
         Raises
         -------
-        AttributeError
-            If the record dict can not be made from ``qca_record``, or if the
-            provided ``client`` could not retrieve the initial molecule.
-        KeyError
+        InvalidQCInputError
+            If the input record isn't suitable to be made into an OpenFF Molecule
+        MissingCMILESError
             If the record does not contain the
             ``canonical_isomeric_explicit_hydrogen_mapped_smiles``.
         InvalidConformerError
-            Silent error, if the conformer could not be attached.
+            If the conformer could not be attached.
         """
-
-        # We can accept the Dataset entry record or the dict with JSON encoding
-        # lets get it all in the dict rep
-        if not isinstance(qca_record, dict):
+        # Process input as dict; convert if necessary
+        if not isinstance(qca_object, dict):
             try:
-                qca_record = qca_record.dict(encoding="json")
+                qca_object = qca_object.dict()
             except AttributeError:
                 raise AttributeError(
-                    "The object passed could not be converted to a dict with json encoding"
+                    f"The input object (type {type(qca_object)=} "
+                    "passed could not be converted to a dict."
+                )
+        if "symbols" in qca_object.keys():
+            mol_dicts = [qca_object]
+        else:
+            mol_dicts = [qca_object.get("molecule")]
```

Comment:
I can't tell if these are actually dictionaries or just objects that quack like them - if they're lists of QCPortal objects I suggust a variable name like `mol_objs`

Reply:
I struggled with how to code this part clearly - At this point, `mol_dicts` are guaranteed to be dicts, since we dictified `qca_object` above, and that will have recursively dictified any molecules it owns.